File: bindings/python/python/maxt/examples/binance_provider.py
"""Use public Binance provider-specific Spot and USD-M reads."""

from __future__ import annotations

import asyncio

from maxt import BinanceAdapter, Exchange, Market


async def main() -> None:
    spot_market = Market.spot(Exchange.BINANCE, "BTC", "USDT")
    spot = BinanceAdapter.spot()
    average = await spot.spot_average_price(spot_market)
    filters = await spot.spot_symbol_filters(spot_market)
    exchange = await spot.spot_exchange_info()
    print(f"{average.minutes}-minute average={average.price}; tick={filters.tick_size}")
    print(f"Spot symbols: {len(exchange.symbols)}")

    futures = BinanceAdapter.usd_m_futures()
    metadata = await futures.usd_m_exchange_info()
    print(f"USD-M symbols: {len(metadata.symbols)}")


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/dart/example/binance_provider.dart
import 'package:maxt/maxt.dart';

Future<void> main() async {
  await Maxt.initialize();
  try {
    final market = Market.spot(Exchange.binance, 'BTC', 'USDT');
    final spot = BinanceAdapter.spot();
    final average = await spot.spotAveragePrice(market);
    final filters = await spot.spotSymbolFilters(market);
    final exchange = await spot.spotExchangeInfo();
    print(
      '${average.minutes}-minute average=${average.price}; tick=${filters.tickSize}',
    );
    print('Spot symbols: ${exchange.symbols.length}');

    final futures = BinanceAdapter.usdMFutures();
    final metadata = await futures.usdMExchangeInfo();
    print('USD-M symbols: ${metadata.symbols.length}');
  } finally {
    await Maxt.dispose();
  }
}
